### PR TITLE
[MM-32343] Environment SubSection

### PR DESCRIPTION
**Pull request by maxerenberg**
#### Summary
This PR adds subsections in the system console for ENVIRONMENT permissions.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-32343

#### Webapp PR
https://github.com/mattermost/mattermost-webapp/pull/7622

### CODE REVIEW

**Review by BenCookie95 — APPROVED** [review]
LGTM!

File: src/constants/permissions.ts [hunk 1]
```diff
@@ -195,8 +255,16 @@ values.SYSCONSOLE_ANCILLARY_PERMISSIONS = {
         values.READ_JOBS,
         values.DOWNLOAD_COMPLIANCE_EXPORT_RESULT,
     ],
-    [values.SYSCONSOLE_READ_ENVIRONMENT]: [
```
Comment (hahmadia) [inline src/constants/permissions.ts:198]: Tagged you in on one of my PR's that gets rid of these SYSCONSOLE Ancillary on the frontend. Will help avoid having to manage this.

File: src/constants/permissions_sysconsole.ts [hunk 2]
```diff
@@ -16,19 +16,46 @@ export const RESOURCE_KEYS = {
...
+    [RESOURCE_KEYS.ENVIRONMENT.PUSH_NOTIFICATION_SERVER]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_PUSH_NOTIFICATION_SERVER, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_PUSH_NOTIFICATION_SERVER],
+    [RESOURCE_KEYS.ENVIRONMENT.HIGH_AVAILABILITY]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_HIGH_AVAILABILITY, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_HIGH_AVAILABILITY],
+    [RESOURCE_KEYS.ENVIRONMENT.RATE_LIMITING]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_RATE_LIMITING, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_RATE_LIMITING],
+    [RESOURCE_KEYS.ENVIRONMENT.LOGGING]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_LOGGING, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_LOGGING],
+    [RESOURCE_KEYS.ENVIRONMENT.SESSION_LENGTHS]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_SESSION_LENGTHS, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_SESSION_LENGTHS],
+    [RESOURCE_KEYS.ENVIRONMENT.PERFORMANCE_MONITORING]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_PERFORMANCE_MONITORING, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_PERFORMANCE_MONITORING],
+    [RESOURCE_KEYS.ENVIRONMENT.DEVELOPER]: [Permissions.SYSCONSOLE_READ_ENVIRONMENT_DEVELOPER, Permissions.SYSCONSOLE_WRITE_ENVIRONMENT_DEVELOPER],
```
Comment (hahmadia) [inline src/constants/permissions_sysconsole.ts:52]: Can we move these below user management (to follow the ordering on the system console itself)

**Review by hahmadia — COMMENTED** [review]
Looks really good! I like what I see 👍 very minor change but great work.

**Review by sbishel — APPROVED** [review]

**Review by hahmadia — APPROVED** [review]
Hi Max,

These look great. Please remember to merge the latest SectionsToSubSection changes and remove all the ancillary permissions from your code. 

If any changes happen in the Server code, feel free to re-assign me to re-review this PR if any changes need to be made.

Thank you!

**Comment by sbishel** [comment]
Performed QA on this set of PRs. Test write, read and no access for each subsection and ability to perform ancillary actions.

**Pull request merged**